SearchableSnapshotsPersistentCacheIntegTests. testCacheSurviveRestart failure

**Build scan**:
https://gradle-enterprise.elastic.co/s/xy7c3vznahvoc
**Repro line**:
```
./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.cache.SearchableSnapshotsPersistentCacheIntegTests.testCacheSurviveRestart" -Dtests.seed=9F0422EA69B04B29 -Dtests.security.manager=true -Dtests.locale=ga -Dtests.timezone=Asia/Yakutsk -Druntime.java=11
```
**Reproduces locally?**:
No, ran it with the same seed, and many different seeds but with other params being the same
**Applicable branches**:
master
**Failure excerpt**:
```
org.elasticsearch.xpack.searchablesnapshots.cache.SearchableSnapshotsPersistentCacheIntegTests > testCacheSurviveRestart FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: <0L> |  
  | but: was <12L> |  
  | at __randomizedtesting.SeedInfo.seed([9F0422EA69B04B29:851033F89C267B6E]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.xpack.searchablesnapshots.cache.SearchableSnapshotsPersistentCacheIntegTests.testCacheSurviveRestart(SearchableSnapshotsPersistentCacheIntegTests.java:155)
```

Another failure: https://gradle-enterprise.elastic.co/s/jpxhkvlcqq3ze/console-log?task=:x-pack:plugin:searchable-snapshots:internalClusterTest

```

org.elasticsearch.xpack.searchablesnapshots.cache.SearchableSnapshotsPersistentCacheIntegTests > testCacheSurviveRestart FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: <0L> |  
  | but: was <10L> |  
  | at __randomizedtesting.SeedInfo.seed([98F3DDD081D6BC00:82E7CCC274408C47]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923)
```

--- a/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/SearchableSnapshotsPersistentCacheIntegTests.java
+++ b/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/SearchableSnapshotsPersistentCacheIntegTests.java
@@ -7,13 +7,13 @@
 package org.elasticsearch.xpack.searchablesnapshots.cache;
 
 import org.apache.lucene.document.Document;
-import org.apache.lucene.util.LuceneTestCase;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.cluster.node.DiscoveryNode;
 import org.elasticsearch.cluster.node.DiscoveryNodes;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.unit.ByteSizeUnit;
 import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.index.Index;
 import org.elasticsearch.index.IndexService;
 import org.elasticsearch.index.shard.ShardPath;
@@ -39,7 +39,6 @@ import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.notNullValue;
 
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/66278")
 public class SearchableSnapshotsPersistentCacheIntegTests extends BaseSearchableSnapshotsIntegTestCase {
 
     @Override
@@ -48,6 +47,8 @@ public class SearchableSnapshotsPersistentCacheIntegTests extends BaseSearchable
             .put(super.nodeSettings(nodeOrdinal))
             // ensure the cache is definitely used
             .put(CacheService.SNAPSHOT_CACHE_SIZE_SETTING.getKey(), new ByteSizeValue(1L, ByteSizeUnit.GB))
+            // to make cache synchronization predictable
+            .put(CacheService.SNAPSHOT_CACHE_SYNC_INTERVAL_SETTING.getKey(), TimeValue.timeValueHours(1L))
             .build();
     }
 
@@ -141,19 +142,21 @@ public class SearchableSnapshotsPersistentCacheIntegTests extends BaseSearchable
             }
         });
 
-        persistentCache = internalCluster().getInstance(CacheService.class, dataNode).getPersistentCache();
-        assertThat(persistentCache.getNumDocs(), equalTo((long) cacheFiles.size()));
+        cacheService = internalCluster().getInstance(CacheService.class, dataNode);
+        persistentCache = cacheService.getPersistentCache();
         ensureGreen(restoredIndexName);
 
         cacheFiles.forEach(cacheFile -> assertTrue(cacheFile + " should have survived node restart", Files.exists(cacheFile)));
+        assertThat("Cache files should be repopulated in cache", persistentCache.getNumDocs(), equalTo((long) cacheFiles.size()));
 
         assertAcked(client().admin().indices().prepareDelete(restoredIndexName));
 
-        assertBusy(() -> cacheFiles.forEach(cacheFile -> assertFalse(cacheFile + " should have been cleaned up", Files.exists(cacheFile))));
-        cacheService = internalCluster().getInstance(CacheService.class, dataNode);
+        assertBusy(() -> {
+            cacheFiles.forEach(cacheFile -> assertFalse(cacheFile + " should have been cleaned up", Files.exists(cacheFile)));
+            assertTrue(internalCluster().getInstance(CacheService.class, dataNode).getPersistentCache().hasDeletions());
+        });
         cacheService.synchronizeCache();
 
-        persistentCache = cacheService.getPersistentCache();
         assertThat(persistentCache.getNumDocs(), equalTo(0L));
     }
 }

